core/src/main/java/com/linecorp/armeria/common/logging/ExportGroupBuilder.java

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.logging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Splitter;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+
+import io.netty.util.AsciiString;
+import io.netty.util.AttributeKey;
+
+/**
+ * Builds a new {@link ExportGroup}.
+ */
+public final class ExportGroupBuilder {
+
+    private static final String PREFIX_REQ_HEADERS = "req.headers.";
+    private static final String PREFIX_RES_HEADERS = "res.headers.";
+
+    static final String PREFIX_ATTRS = "attrs.";
+    private static final String ATTR_NAMESPACE = "attr:";
+
+    private static final Splitter KEY_SPLITTER = Splitter.on(',').trimResults();
+
+    @Nullable
+    private String prefix;
+    private final Set<ExportEntry<BuiltInProperty>> builtIns;
+    private final Set<ExportEntry<AttributeKey<?>>> attrs;
+    private final Set<ExportEntry<AsciiString>> reqHeaders;
+    private final Set<ExportEntry<AsciiString>> resHeaders;
+
+    /**
+     * Returns a new {@link ExportGroupBuilder}.
+     */
+    public ExportGroupBuilder() {
+        builtIns = new HashSet<>();
+        attrs = new HashSet<>();
+        reqHeaders = new HashSet<>();
+        resHeaders = new HashSet<>();
+    }
```

Comment:
I think we need to separate the core API from Logback integration. Could we ensure this builder API doesn't expose the default constructor and doesn't have the `set` prefix? Instead, we could define some additional public class that provides a bridge between this class and Logback configuration, e.g.

```java
private ExportGroupBuilder builder = ExportGroup.builder();
public void setPrefix(String prefix) {
  builder.prefix(prefix);
}
```